llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py

```diff
@@ -22,34 +24,43 @@ def __init__(
         self,
         model,
         top_n: int = 10,
-        with_retrieval: bool = False,
+        with_retrieval: Optional[bool] = False,
+        step_size: Optional[int] = 10,
     ):
-        try:
-            model_enum = ModelType(model.lower())
-        except ValueError:
-            raise ValueError("Unsupported model type. Please use 'vicuna' or 'zephyr'.")
+        model_name = model.lower()
         from rank_llm.result import Result
 
         self._result = Result
 
-        if model_enum == ModelType.VICUNA:
```

Comment:
Curious why we removed the enum? Seemed helpful for validation

Reply:
we'd need to add another parameter `gpt-type`, or continuously update the enums to match the available openai models. 

I wanted to keep the API clean. Tho thinking more, i think its ok adding another optional parameter, if that sounds good.